mm2src/coins/eth.rs

```diff
@@ -6837,7 +6866,8 @@ impl ParseCoinAssocTypes for EthCoin {
     }
 
     fn parse_pubkey(&self, pubkey: &[u8]) -> Result<Self::Pubkey, Self::PubkeyParseError> {
-        HtlcPubKey::from_slice(pubkey).map_to_mm(EthAssocTypesError::from)
+        let pubkey = PublicKey::from_slice(pubkey)?;
+        Ok(Public::from_slice(&pubkey.serialize_uncompressed()[1..65]))
```

Comment:
hmm I decided not to do the refactoring in nft swap tests right now, so they still use legacy `derive_htlc_pubkey` in tests. And I also use `parse_pubkey` in nft tests.
So right now `fn parse_pubkey` is able to parse `ethereum_types Public` type from slices provided by `keys::public`
Check `send_nft_maker_payment` as an example
https://github.com/KomodoPlatform/komodo-defi-framework/blob/dab5bd996fe5620623aac5b2f3cd4016c3c27a20/mm2src/mm2_main/tests/docker_tests/eth_docker_tests.rs#L1300-L1317

I left this todo in next pr https://github.com/KomodoPlatform/komodo-defi-framework/pull/2169/files#diff-b1bb9a17c62e6d79bb33d5b8a0db3954cc6b1763cbd95c02977023e39f0ea44cR60-R62 as in eth docker swaps I use `derive_htlc_pubkey_v2`

I suppose current `parse_pubkey` should be able to parse ethereum `Public` not only from `derive_htlc_pubkey` but also from `derive_htlc_pubkey_v2`. Yeah kinda wierd to derive ethereum Public and and in `parse_pubkey`maker `secp256k1::PublicKey` to make ethereum Public again.
But I think i can use `parse_pubkey` to solve the todo above.

---
But about coins specific vs same pubkey note
>  I need to look into if using the coins specific pubkey is better or using the same pubkey for all coins as it was done for v1 is still the best option. 

If we choose `coins specific` approach, then in `parse_pubkey` we have to support both legacy derive and derive_v2. However we need to do this as in swap_v2 we still use legacy `derive_htlc_pubkey` together with `derive_htlc_pubkey_v2`
![Screenshot 2024-07-20 at 08 31 15](https://github.com/user-attachments/assets/b1a4a92c-722a-4e21-9506-df98519dc6d7)

Well, I see, maybe `derive_htlc_pubkey_v2` was supposed to generate the same htlc pubkey type as legacy, just it should be `Self::Pubkey`, not slice. But if we use choose the same htlc pubkey, why do we need this `Self::Pubkey` feature :) ?

Reply:
I provided new `derive_htlc_pubkey_v2_bytes` function and used it instead of legacy `derive_htlc_pubkey` in swap v2 rs files. 
Also moved v2 derive htlc pubkey funcs in separate Common Ops trait as both maker and taker coins need `derive_htlc_pubkey_v2_bytes` function

done in this commit https://github.com/KomodoPlatform/komodo-defi-framework/pull/2129/commits/28bc18ac36604a69690693b432ad5ec50783f0f4